Update mimalloc to v2.2.3

dschuff opened the pull request:
This brings in revision 4aae566191b9443d53995245b637ce28d617710a,
needed to build on Windows.

kripken commented:
Sounds good to update, but looks like Alpine is failing on that revision? Maybe we should wait for a stable release?

dschuff commented:
I had thought that this was a stable release (the 2.2.3 tag).
I will double check (GH really does not make it easy to tell exactly what the old a new revisions are for submodules).

kleisauke commented:
The compilation failure on Alpine Linux ought to be fixed with commit https://github.com/microsoft/mimalloc/commit/341149391fee496790a7fa916b1fd3fdd0cce1a1, which is included in mimalloc v2.2.4 (https://github.com/microsoft/mimalloc/commit/fbd8b99c2b828428947d70fdc046bb55609be93e).

dschuff commented:
Updated to 2.2.4. I don't think we necessarily need to have this (on emscripten-releases we've just updated the Linux build to use mimalloc because of a performance issue) but updating seems good in general.
